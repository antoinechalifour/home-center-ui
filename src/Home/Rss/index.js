import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { graphql } from 'react-apollo'
import gql from 'graphql-tag'
import Card, * as card from 'ui/Card'
import Section, * as section from 'Home/Section'

const Feed = styled.ul`
  padding: 12px;
  display: flex;

  @media (min-width: 860px) {
    flex-direction: column;
  }
`

const Item = styled.li`
  flex: 0 0 90%;

  & + & {
    margin-left: 12px;
    margin-right: 12px;
  }

  @media (min-width: 860px) {
    & + & {
      margin-left: 0;
      margin-right: 0;
      margin-top: 8px;
    }
  }
`

const ItemLink = styled.a`
  display: block;
  text-decoration: none;
  color: inherit;

  span {
    display: block;
  }
`

const ItemTitle = styled.span`
`

const ItemMeta = styled.span`
  font-size: 80%;
  opacity: .54;
`

const Rss = ({ data }) => {
  return (
    <Section>
      <section.Title>News</section.Title>
      <section.Content>
        {data.loading && <section.Loader />}
        {data.feed &&
          <Feed>
            {data.feed.map(({ title, link, date, source }) => (
              <Item key={title}>
                <Card>
                  <card.Content>
                    <ItemLink href={link} target='_blank' rel='noopener'>
                      <ItemTitle>{title}</ItemTitle>
                      <ItemMeta>{source}</ItemMeta>
                    </ItemLink>
                  </card.Content>
                </Card>
              </Item>
            ))}
          </Feed>}
      </section.Content>
    </Section>
  )
}

Rss.propTypes = {
  data: PropTypes.shape({
    loading: PropTypes.bool.isRequired,
    feed: PropTypes.arrayOf(
      PropTypes.shape({
        title: PropTypes.string.isRequired,
        link: PropTypes.string.isRequired,
        date: PropTypes.string.isRequired,
        source: PropTypes.string.isRequired
      })
    )
  }).isRequired
}

export default graphql(
  gql`
  query RssQuery {
    feed {
      title,
      link,
      date,
      source
    }
  }
`
)(Rss)
