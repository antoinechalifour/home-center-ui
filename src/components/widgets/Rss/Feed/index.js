import { graphql } from 'react-apollo'
import { getRssFeed } from 'queries/rss'
import Feed from './Feed'

export default graphql(getRssFeed)(Feed)
