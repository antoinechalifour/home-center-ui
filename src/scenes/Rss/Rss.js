import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Card, * as card from 'ui/Card'
import Loader from 'ui/Loader'
import Col from 'ui/Col'

export default function Rss ({ data }) {
  return (
    <Col align='center'>
      {data.loading && <Loader />}
      {data.feed &&
        <Feed>
          {data.feed.map(({ title, link, date, source }) => (
            <Item key={title}>
              <Card>
                <card.Content>
                  <ItemLink href={link} target='_blank' rel='noopener'>
                    <span>{title}</span>
                    <ItemMeta>{source}</ItemMeta>
                  </ItemLink>
                </card.Content>
              </Card>
            </Item>
          ))}
        </Feed>}
    </Col>
  )
}

Rss.propTypes = {
  data: PropTypes.shape({
    loading: PropTypes.bool.isRequired,
    feed: PropTypes.arrayOf(
      PropTypes.shape({
        title: PropTypes.string.isRequired,
        link: PropTypes.string.isRequired,
        date: PropTypes.string.isRequired,
        source: PropTypes.string.isRequired
      })
    )
  }).isRequired
}

const Feed = styled.ul`
  display: flex;
  width: 100%;

  @media (min-width: 860px) {
    flex-direction: column;
  }
`

const Item = styled.li`
  flex: 0 0 90%;

  & + & {
    margin-left: 12px;
    margin-right: 12px;
  }

  @media (min-width: 860px) {
    & + & {
      margin-left: 0;
      margin-right: 0;
      margin-top: 8px;
    }
  }
`

const ItemLink = styled.a`
  display: block;
  text-decoration: none;
  color: inherit;

  span {
    display: block;
  }
`

const ItemMeta = styled.span`
  font-size: 80%;
  opacity: .54;
`
