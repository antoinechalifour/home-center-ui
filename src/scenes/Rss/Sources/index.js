import React, { Fragment } from 'react'
import Sources from './Sources'
import NewSource from './NewSource'

export default function SourcesPane () {
  return (
    <Fragment>
      <Sources />
      <NewSource />
    </Fragment>
  )
}
