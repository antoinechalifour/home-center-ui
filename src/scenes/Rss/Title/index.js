import { graphql } from 'react-apollo'
import { addSource } from 'queries/rss'
import Title from './Title'

const addSourceOptions = {
  name: 'addSource',
  options: {
    refetchQueries: ['GetRssQuery']
  }
}

export default graphql(addSource, addSourceOptions)(Title)
