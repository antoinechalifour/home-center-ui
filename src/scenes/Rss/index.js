import { graphql, compose } from 'react-apollo'
import { getRssFeed, deleteSource } from 'queries/rss'
import Rss from './Rss'

const deleteSourceOptions = {
  name: 'deleteSource',
  options: {
    refetchQueries: ['GetRssQuery']
  }
}

export default compose(
  graphql(getRssFeed),
  graphql(deleteSource, deleteSourceOptions)
)(Rss)
