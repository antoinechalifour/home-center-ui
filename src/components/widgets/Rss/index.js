import React from 'react'
import styled from 'styled-components'
import Feed from './Feed'

export default function Rss () {
  return (
    <div>
      <Title>Highlights</Title>
      <Content>
        <Feed />
      </Content>
    </div>
  )
}

const Title = styled.div`
  background: ${({ theme }) => theme.colors.primary};
  color: #fff;
  padding: 12px;
  text-align: center;
  text-transform: uppercase;
`

const Content = styled.div`
  padding: 0 12px;
`
