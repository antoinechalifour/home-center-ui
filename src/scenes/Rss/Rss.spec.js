import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import Rss from './Rss'

const renderer = new ShallowRenderer()

describe('scenes/Rss/Rss', () => {
  it('Should render a loader when the data is loading', () => {
    const props = { data: { loading: true } }
    const tree = renderer.render(<Rss {...props} />)

    expect(tree).toMatchSnapshot()
  })

  it('Should render the feed otherwise', () => {
    const props = {
      data: {
        loading: false,
        feed: [
          {
            date: 'Tue, 26 Dec 2017 09:46:38 +0100',
            link: 'http://fizz.buzz',
            source: 'FizzBuzz Magazine',
            title: 'Fizz! Buzz!'
          },
          {
            date: 'Tue, 26 Dec 2017 09:42:38 +0100',
            link: 'http://foo.bar',
            source: 'FooBar Magazine',
            title: 'Foo! Bar!'
          }
        ]
      }
    }
    const tree = renderer.render(<Rss {...props} />)

    expect(tree).toMatchSnapshot()
  })
})
