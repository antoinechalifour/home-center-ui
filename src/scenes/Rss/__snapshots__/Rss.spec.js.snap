// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scenes/Rss/Rss Should render a loader when the data is loading 1`] = `
<Flex(column)
  align="center"
  justify="flex-start"
>
  <Loader />
</Flex(column)>
`;

exports[`scenes/Rss/Rss Should render the feed otherwise 1`] = `
<Flex(column)
  align="center"
  justify="flex-start"
>
  <styled.ul>
    <styled.li>
      <styled.div>
        <styled.div>
          <styled.a
            href="http://fizz.buzz"
            rel="noopener"
            target="_blank"
          >
            <span>
              Fizz! Buzz!
            </span>
            <styled.span>
              FizzBuzz Magazine
            </styled.span>
          </styled.a>
        </styled.div>
      </styled.div>
    </styled.li>
    <styled.li>
      <styled.div>
        <styled.div>
          <styled.a
            href="http://foo.bar"
            rel="noopener"
            target="_blank"
          >
            <span>
              Foo! Bar!
            </span>
            <styled.span>
              FooBar Magazine
            </styled.span>
          </styled.a>
        </styled.div>
      </styled.div>
    </styled.li>
  </styled.ul>
</Flex(column)>
`;
